### PR TITLE
Fix for fmt > 8.0

### LINKED ISSUE #2383: Build failure with fmt > 8.0
After upgrading fmt (8.0.1-2 -> 8.1.1-1) there is now a build error:

```
In file included from /usr/include/spdlog/fmt/fmt.h:25,
                 from /usr/include/spdlog/common.h:36,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/util/logger.h:35,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/common.h:36,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/util/timer.h:39,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.h:38,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.cc:32:
/usr/include/fmt/core.h: In instantiation of 'constexpr fmt::v8::detail::value<Context> fmt::v8::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; fmt::v8::detail::type <anonymous> = fmt::v8::detail::type::custom_type; T = ScanMode&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]':
/usr/include/fmt/core.h:1855:77:   required from 'constexpr fmt::v8::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {ScanMode&}; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {ScanMode}]'
/usr/include/fmt/core.h:1872:38:   required from 'constexpr fmt::v8::format_arg_store<Context, fmt::v8::remove_cvref_t<Args>...> fmt::v8::make_format_args(Args&& ...) [with Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {ScanMode&}]'
/usr/include/fmt/core.h:3119:44:   required from 'std::string fmt::v8::format(fmt::v8::format_string<T ...>, T&& ...) [with T = {ScanMode&}; std::string = std::__cxx11::basic_string<char>; fmt::v8::format_string<T ...> = fmt::v8::basic_format_string<char, ScanMode&>]'
/build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.cc:124:5:   required from here
/usr/include/fmt/core.h:1728:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1728 |       formattable,
      |       ^~~~~~~~~~~
make[2]: *** [CMakeFiles/libgerbera.dir/build.make:230: CMakeFiles/libgerbera.dir/src/content/autoscan.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/libgerbera.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

```

### PR DESCRIPTION
**Pull request by kyak**
Fixes #2383 

See also https://github.com/fmtlib/fmt/issues/2704 and https://github.com/yuzu-emu/yuzu/commit/b3308830b217cbdd9638b11e8660d079e16b3f01

### CODE REVIEW

File: src/content/autoscan.cc [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #include "content_manager.h"
 #include "database/database.h"
 
+#include <fmt/ostream.h>
```
Comment (KarlStraussberger) [inline src/content/autoscan.cc:37]: Why is it necessary. It must be before the other includes.

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

File: src/content/autoscan.cc (hunk 1 above)
Comment (kyak) [reply src/content/autoscan.cc]: Sorry, that was a bullshit commit. This should be a proper fix.

**Comment by whyman** [comment]
Looks good, just need the formatting fix

**Review by whyman — APPROVED** [review]

File: src/content/autoscan.h [hunk 2]
```diff
@@ -34,6 +34,8 @@
 #define __AUTOSCAN_H__
 
 #include <mutex>
+#include <type_traits>
+#include <fmt/format.h>
```
Comment (KarlStraussberger) [inline src/content/autoscan.h:38]: Alphabetical order is required as well

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
Great change. Please move it to `logger.h`

**Comment by neheb** [comment]
Instead of this PR there's also a to_underlying function.

**Review by KarlStraussberger — APPROVED** [review]

**Comment by whyman** [comment]
Thanks!

**Pull request merged**